common4j/src/main/com/microsoft/identity/common/java/providers/nativeauth/requests/signin/SignInTokenRequest.kt

```diff
@@ -0,0 +1,175 @@
+            ArgUtils.validateNonNullArg(credentialToken, "credentialToken")
+            ArgUtils.validateNonNullArg(clientId, "clientId")
+            ArgUtils.validateNonNullArg(challengeType, "challengeType")
+            ArgUtils.validateNonNullArg(requestUrl, "requestUrl")
+            ArgUtils.validateNonNullArg(headers, "headers")
+
+
+            return SignInTokenRequest(
+                parameters = NativeAuthRequestSignInTokenParameters(
+                    password = password,
+                    credentialToken = credentialToken,
+                    clientId = clientId,
+                    grantType = NativeAuthConstants.GrantType.PASSWORD,
+                    challengeType = challengeType,
+                    scope = scopes?.joinToString(" ")
+                ),
+                requestUrl = URL(requestUrl),
+                headers = headers,
+            )
+        }
+
+        /**
+         * Returns a request object using the provided parameters.
+         * The request URL and headers passed will be set directly.
+         * The NativeAuthRequestSignInChallengeParameters object will be populated with the provided username and SLT, and the grant type will be set to "slt".
+         *
+         * Parameters outside of scopes and challengeType that are null or empty will throw a ClientException.
+         * @see com.microsoft.identity.common.java.exception.ClientException
+         */
+        fun createSltTokenRequest(
+            signInSlt: String,
+            clientId: String,
+            username: String,
+            scopes: List<String>? = null,
+            challengeType: String? = null,
+            requestUrl: String,
+            headers: Map<String, String?>
+        ): SignInTokenRequest {
+            // Check for empty Strings and empty Maps
+            ArgUtils.validateNonNullArg(signInSlt, "signInSlt")
+            ArgUtils.validateNonNullArg(clientId, "clientId")
+            ArgUtils.validateNonNullArg(username, "username")
+            ArgUtils.validateNonNullArg(challengeType, "challengeType")
+            ArgUtils.validateNonNullArg(requestUrl, "requestUrl")
+            ArgUtils.validateNonNullArg(headers, "headers")
+
+            return SignInTokenRequest(
+                parameters = NativeAuthRequestSignInTokenParameters(
+                    signInSlt = signInSlt,
+                    clientId = clientId,
+                    username = username,
+                    grantType = NativeAuthConstants.GrantType.SLT,
+                    challengeType = challengeType,
+                    scope = scopes?.joinToString(" ")
+                ),
+                requestUrl = URL(requestUrl),
+                headers = headers
+            )
+        }
+    }
+
+    data class NativeAuthRequestSignInTokenParameters(
+        val username: String? = null,
+        val password: String? = null,
+        val oob: String? = null,
+        @SerializedName("nca") val nca: Int? = null,
```

Comment:
What is nca?

Reply:
I see, maybe let's include a comment somewhere in this class to explain that. Javadoc could be used for this as well

Reply:
Added